### DISCUSSION TITLE
Static pages created at different times

### DISCUSSION THREAD

**Discussion by matheus-rosin** [vercel/next.js#17389] [Help] [answered]
I have a project that uses static pages through `next export`. But there is a lot of pages, so I can't export all of them each time something changes.

Modifications are, so to speak, saved, and a cron reads these from time to time, and creates a dynamic `exportPathMap`, so that only what is really needed will be exported.

Then the static files are sent to the same place where all others unmodified pages are too.

The trouble is that, because the unmodified pages and the modified ones belong to different build times and bundles, when I click at a link (that uses `Link` component) that would go from, say, an unmodified to a modified page, it does not occur through JS, so the window refreshes (i.e. does not behave as a SPA).

Anything to avoid that?

Thanks in advance 😺

**Comment by jamesmosier** [comment, marked as answer]
In order for `next/link` to work the pages must share the same Context Provider which means they came from the same build. Otherwise Next does not know they are related.

**Reply by jamesmosier** [reply]
> Right... is there any workaround so that I can get this thing working, just in my case? I mean, some code modification I should make in internals locally

There are no internals or Next options you can change to resolve this. 

Although not compatible with `next export`, I think it might be worth looking at the `fallback: true` option with `getStaticProps`. It would allow you to just build a subset of pages and the rest are built on demand: https://nextjs.org/docs/basic-features/data-fetching#fallback-true

Another options might be the `revalidate` option (again not `next export` compatible) but that would solve your issue of having to run another build whenever some "modifications" are made. Assuming that the "modifications" are from an API or CMS.

Aside from those options I'm not personally aware of any other workaround. But someone else might have an idea 😄

**Reply by matheus-rosin** [reply]
So I figured out that if I execute `next build` only once and afterwards execute just `next export` (thus preserving the build), a lot of "minor modifications" can be handled; for the rest there is no problem to build it all again.

Yet I am thinking of a mixed things like `fallback: true`.

Thanks!!

**Comment by matheus-rosin** [comment]
Right... is there any workaround so that I can get this thing working, just in my case? I mean, some code modification I should make in internals locally
